packages/components/LIMITATIONS.md

```diff
@@ -145,12 +133,11 @@ Synergy components make heavy use of [ShadowDOM](https://developer.mozilla.org/e
 
 ### Proposed Solution
 
-Try to match on the next `<syn-nav-item>` with the closest selector. This will also match for slotted elements.
+Use `currentTarget` instead, which aims for the element on which the listener is attached on instead of the event origin.
```

Comment:
This does not always work as the main event may be added to another element higher up in the DOM. For example, we are using a listener like the following to make it work:

```typescript
    const root = document.querySelector('syn-side-nav');
    root?.addEventListener('click', (e: Event) => {
      e.preventDefault();
      e.stopPropagation();

      console.log(e.currentTarget, e.target, e.target.closest('syn-nav-item'));

      const clickedNavItem = (e.target as HTMLElement).closest('syn-nav-item') as SynNavItem;

      if (clickedNavItem) {
        history.push(history.createHref(clickedNavItem.href));
      }
    });
  });
```

In this case, `e.currentTarget` will point to the `<syn-side-nav>`. It is much easier in my opinion to select this via the `closest` selector as people will not always add the event to the associated `<syn-nav-item>` (e.g. for performance-reasons).

What do you think about adding your variant, too as another solution. I can easily see people using both versions (e.g. adding the listener directly to the `<syn-nav-item>` and to the `<syn-side-nav>`.

Reply:
fixed